controller/userController.js

```diff
@@ -31,6 +53,30 @@ const login = (req, res) => {
     });
 };
 
+// TODO: Discuss about potential token refresh
+// For mobile apps users ususally login only once so JWT shouldn't have any expiration date
+// However we should be able to revoke a JWT (e.g. users phone has been stolen)
+const token = (req, res) => {
+    const data = req.body;
+    // If refresh token exists
+    if ((data.refreshToken) && (data.refreshToken in tokenList)) {
+        const user = { email: data.email, active: true }
+        User.findOne(user).then(user => {
+            if (user) {
+                // Create new token
+                const token = jwt.sign({ id: user._id }, AppConfig.SECRET, { expiresIn: AppConfig.TOKEN_LIFESPAN });
+                const response = { "token": token };
+                // Update token in the list
+                tokenList[data.refreshToken].token = token
+                res.status(HttpStatus.OK).json(response);
+            } else {
+                res.status(HttpStatus.NOT_FOUND).send("User not found");
```

Comment:
Again should be generalized. This time the return value is not even a JSON object, but a block of text.